### PR TITLE
Feature/copytensor

### LINKED ISSUE #4647: Remove `CopyFrom` and `CopyFromVector` from Tensor interface and make them global functions
Remove `CopyFrom` and `CopyFromVector` from Tensor interface and make them global functions

### PR DESCRIPTION
**Pull request by dzhwinter**
fix #4647

This PR contains
1. fix #4647 , move utility function out of Tensor.
2. add a CopyToVector Interface. 
3. Delete unused tensorarray, dynamic_recurrent_rnn. We have another implementation. @Superjom 
4. fix the gru_unit_op in debug mode.

### CODE REVIEW

File: paddle/framework/CMakeLists.txt [hunk 1]
```diff
@@ -5,8 +5,13 @@ cc_library(ddim SRCS ddim.cc DEPS eigen3)
 cc_test(ddim_test SRCS ddim_test.cc DEPS ddim)
 nv_test(dim_test SRCS dim_test.cu DEPS ddim)
 
+# cc_library(tensor SRCS tensor.cc tensor_util.cc DEPS ddim place paddle_memory device_context)
```
Comment (typhoonzero) [inline paddle/framework/CMakeLists.txt]: No commented out codes please.

Comment (dzhwinter) [reply paddle/framework/CMakeLists.txt]: Done.

File: python/paddle/v2/framework/tests/test_elementwise_mod_op.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import unittest
```
Comment (typhoonzero) [inline python/paddle/v2/framework/tests/test_elementwise_mod_op.py]: should remove this file since or put under `v2/fluid/test`

Comment (dzhwinter) [reply python/paddle/v2/framework/tests/test_elementwise_mod_op.py]: I have another PR in of https://github.com/PaddlePaddle/Paddle/pull/5454. which I can move this file to the correct place in that PR.  
This PR is so huge and fragile, I think we can mark that and do it later.

Comment (typhoonzero) [reply python/paddle/v2/framework/tests/test_elementwise_mod_op.py]: OK

**Comment by jacquesqiao** [comment]
please clear code before merge if code is too old

**Pull request merged**